hardware_interface/include/hardware_interface/actuator.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__ACTUATOR_HPP_
+#define HARDWARE_INTERFACE__ACTUATOR_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/component_interfaces/actuator_interface.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class Actuator final
+{
+public:
+  explicit Actuator(std::unique_ptr<ActuatorInterface> impl)
+  : impl_(std::move(impl))
+  {}
+
+  ~Actuator() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  hardware_interface_ret_t configure(const ComponentInfo & actuator_info)
+  {
+    return impl_->configure(actuator_info);
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  hardware_interface_ret_t start()
+  {
+    return impl_->start();
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  hardware_interface_ret_t stop()
+  {
+    return impl_->stop();
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  bool is_started() const
+  {
+    return impl_->is_started();
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  hardware_interface_ret_t read(double & data)
+  {
+    return impl_->read(data);
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  hardware_interface_ret_t write(const double & data)
+  {
+    return impl_->write(data);
+  }
+
+  HARDWARE_INTERFACE_PUBLIC
+  std::string get_interface_name() const
+  {
+    return impl_->get_interface_name();
+  }
+
```

Comment:
Did you intentionally left out the actuator claiming functions?

If not, we should add than fulll interface what I wrote in 54165d3 before merging.

Reply:
I was thinking about having this either in the configure step or do the assignment somewhere generic (maybe in the hardware interface or the actuator). I wonder if each individual implementation (see `robot_control_components`) should have to deal with this.

Reply:
Would‘t than this be the perfect place? I understand that Container for hardware interacts with this class and than Actuator do not have to forward the call to the implementation. Did I misunderstood something?

Reply:
yup. that sounds good to me. The constructor of this class could take a shared list of resources and mark the claimed resources as such.

Reply:
If we claim resources  in constructor, would we have problems with switching controllers?

Reply:
This will be discussed in ros-controls/ros2_control#101